stm32f407Xg: RAM is 128K + 64K CCM

erwango opened the pull request:
stm32f407Xg package was defined with 192K RAM + 64K CCM.
This is wrong as total RAM is 192, including 64K CCM.

Fix RAM size definition

Fixes #14779

erwango committed:
dts: stm32: Fix erroneous RAM sizes when CCM is available

On various stm32 soc packages CCM is available and SRAM size
mentionned in datasheet include CCM.
Though, actual SRAM size defined in dtsi files should not
include CCM sizes.
Fix this on impacted dtsi files.

Fixes #14779

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

KwonTae-young commented:
@erwango

I have tried to find the CPU that contains the CCM.
```
root:zephyr# grep -r ccm dts/arm/st
dts/arm/st/f3/stm32f303Xc.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f3/stm32f303Xc.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f3/stm32f334X8.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f3/stm32f334X8.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f405Xg.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f405Xg.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f429Xi.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f429Xi.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f469Xi.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f469Xi.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f417Xg.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f417Xg.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f417Xe.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f417Xe.dtsi:		compatible = "st,stm32-ccm";
dts/arm/st/f4/stm32f407Xg.dtsi:	ccm0: memory@10000000 {
dts/arm/st/f4/stm32f407Xg.dtsi:		compatible = "st,stm32-ccm";
root:zephyr# 
```

I think the following CPUs need the same modification.

erwango committed:
boards: stm32: Enable CCM and update yaml

Following fix SRAM size to take into account CCM,
enable CCM and update yaml files in boards including fixed soc
dtsi definitions.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@KwonTae-young , you are right, I've update the commits

KwonTae-young commented on boards/arm/stm32f4_disco/stm32f4_disco.yaml:
```diff
@@ -6,6 +6,8 @@ toolchain:
   - zephyr
   - gnuarmemb
   - xtools
+ram: 320
```
128 is not right?
